### PR TITLE
Feature: setup first version of templates

### LINKED ISSUE #1: Setup library with some templates
Initialize the project and setup two or more template.

### PR DESCRIPTION
**Pull request by Zorin95670**
Resolved #1

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,33 @@
 # leto-modelizer-templates-library
+
 Library that contains all default template for leto-modelizer.
+
+## Create a template
+
+1. Create the metadata of yours template in the [index.json](index.json):
```
Comment (k-y0u) [inline README.md]: your templates

File: README.md [hunk 2]
```diff
@@ -1,2 +1,33 @@
...
+    "name": "Web application",
+    "plugin": "terrator-plugin",
+    "type": "model",
+    "description": "Schema of web application.",
+    "url": null,
+    "files": ["main.tf"]
+}
+```
+
+- `key` attribute is the name of folder of your template under the `templates` folder.
```
Comment (k-y0u) [inline README.md]: folder name

File: README.md [hunk 3]
```diff
@@ -1,2 +1,33 @@
...
+- `name` attribute is the display name used in leto-modelizer.
+- `plugin` attribute is the name of plugin used by the template.
```
Comment (k-y0u) [inline README.md]: plugin name

File: README.md [hunk 4]
```diff
@@ -1,2 +1,33 @@
...
+- `type` attribute is the type template, can be `model` or `component`.
```
Comment (k-y0u) [inline README.md]: the type of the template, either `model`, `component` or `project`.

Maybe not `project` in first place....

File: README.md [hunk 5]
```diff
@@ -1,2 +1,33 @@
...
+- `description` attribute is the description used in leto-modelizer to explain the usage of this template.
+- `url` attribute is an url attached to a documentation of this template used in leto-modelizer.
+- `files` attribute is the list of files used in the template.
+
+2. Create the folder of your template with the choosen key you provide in the [index.json](index.json)
+3. Create all you mentioned files in this folder with the code of your template
+4. (Optional) Provided an icon of your template, only an `icon.svg` will be used by leto-modelizer
```
Comment (k-y0u) [inline README.md]: template (only `icon.svg` will be used by leto-modelizer)

File: README.md [hunk 6]
```diff
@@ -1,2 +1,33 @@
...
+5. (Optional) Provided a schema of your template , only an `schema.svg` will be used by leto-modelizer
```
Comment (k-y0u) [inline README.md]: template (only `schema.svg` will be used by leto-modelizer)

**Review by k-y0u — CHANGES_REQUESTED** [review]

File: README.md [hunk 7]
```diff
@@ -1,2 +1,33 @@
 # leto-modelizer-templates-library
+
 Library that contains all default template for leto-modelizer.
```
Comment (justinehell) [inline README.md]: all default templates

**Review by justinehell — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (Zorin95670) [reply README.md]: done

File: README.md (hunk 2 above)
Comment (Zorin95670) [reply README.md]: done

File: README.md (hunk 3 above)
Comment (Zorin95670) [reply README.md]: done

File: README.md (hunk 4 above)
Comment (Zorin95670) [reply README.md]: I don't put project in the documentation because we don't again manage project :)

File: README.md (hunk 5 above)
Comment (Zorin95670) [reply README.md]: done

File: README.md (hunk 6 above)
Comment (Zorin95670) [reply README.md]: done

File: README.md (hunk 7 above)
Comment (Zorin95670) [reply README.md]: done

File: README.md [hunk 8]
```diff
@@ -1,2 +1,33 @@
...
+    "description": "Schema of web application.",
+    "url": null,
+    "files": ["main.tf"]
+}
+```
+
+- `key` attribute is the folder name of your template under the `templates` folder.
+- `name` attribute is the display name used in leto-modelizer.
+- `plugin` attribute is the plugin name used by the template.
+- `type` attribute is the type template, either `model` or `component`.
```
Comment (justinehell) [inline README.md]: `template type`

File: README.md [hunk 9]
```diff
@@ -1,2 +1,33 @@
...
+- `description` attribute is the description used in leto-modelizer to explain the usage of this template.
+- `url` attribute is an url attached to a documentation of this template used in leto-modelizer.
+- `files` attribute is the list of files used in the template.
+
+2. Create the folder of your template with the choosen key you provide in the [index.json](index.json)
```
Comment (justinehell) [inline README.md]: choosen -> chosen key you provide -> provided

File: README.md (hunk 4 above)
Comment (k-y0u) [reply README.md]: you forgot to correct the first half !

File: README.md [hunk 10]
```diff
@@ -1,2 +1,33 @@
...
+3. Create all you mentioned files in this folder with the code of your template
```
Comment (justinehell) [inline README.md]: Create all the files (you) mentioned

File: README.md [hunk 11]
```diff
@@ -1,2 +1,33 @@
...
+4. (Optional) Provided an icon of your template (only `icon.svg` will be used by leto-modelizer)
+5. (Optional) Provided a schema of your template (only `schema.svg` will be used by leto-modelizer)
```
Comment (justinehell) [inline README.md]: Provided -> Provide (because above is `Create` not created)
of -> for ?

**Review by justinehell — CHANGES_REQUESTED** [review]

File: README.md (hunk 8 above)
Comment (Zorin95670) [reply README.md]: done

File: README.md (hunk 9 above)
Comment (Zorin95670) [reply README.md]: done

File: README.md (hunk 10 above)
Comment (Zorin95670) [reply README.md]: done

File: README.md (hunk 11 above)
Comment (Zorin95670) [reply README.md]: done

**Review by k-y0u — APPROVED** [review]

**Review by justinehell — APPROVED** [review]

**Pull request merged**